scripts/monte_carlo.py

```diff
@@ -173,72 +214,96 @@ def monte_carlo_sampling_scipy(
+            "samples"
+        )  # TODO: What is the optimal sampling? Fabian Neumann answered that in "Broad ranges" paper
+        SAMPLING_STRATEGY = OPTIONS.get("sampling_strategy")
+
+        if SAMPLING_STRATEGY == "pydoe2":
+            lh = monte_carlo_sampling_pydoe2(
+                N_FEATURES,
+                SAMPLES,
+                criterion=None,
+                iteration=None,
+                random_state=42,
+                correlation_matrix=None,
+            )
+        if SAMPLING_STRATEGY == "scipy":
+            lh = monte_carlo_sampling_scipy(
+                N_FEATURES,
+                SAMPLES,
+                centered=False,
+                strength=2,
+                optimization=None,
+                seed=42,
+            )
+        if SAMPLING_STRATEGY == "chaospy":
+            lh = monte_carlo_sampling_chaospy(
+                N_FEATURES,
+                SAMPLES,
+                rule="latin_hypercube",
+                seed=42,
+            )
+        scenarios = qmc.scale(lh, L_BOUNDS, U_BOUNDS)
+
+    elif OPTIONS.get("method") == "any_chance_store_test":
+        carrier_no = len(n.stores.carrier.unique())
+        worst_list = L_BOUNDS * carrier_no
+        best_list = U_BOUNDS * carrier_no
+        scenarios = single_best_in_worst_list(worst_list, best_list)
+
+    ### NETWORK MODIFICATION
     ###
-    n = pypsa.Network(snakemake.input[0])
     unc_wildcards = snakemake.wildcards[-1]
     i = int(unc_wildcards[1:])
     j = 0
-    for k, v in MONTE_CARLO_PYPSA_FEATURES.items():
-        # this loop sets in one scenario each "i" feature assumption
-        # k is the config input key "loads_t.p_set"
-        # v is the lower and upper bound [0.8,1.3], that was used for lh_scaled
-        # i, j interation number to pick values of experimental setup
-        # Example: n.loads_t.p_set = network.loads_t.p_set = .loads_t.p_set * lh_scaled[0,0]
-        exec(f"n.{k} = n.{k} * {lh_scaled[i,j]}")
-        logger.info(f"Scaled n.{k} by factor {lh_scaled[i,j]} in the {i} scenario")
-        j = j + 1
+
+    if OPTIONS.get("method") == "global_sensitivity":
+        for k, v in PYPSA_FEATURES.items():
+            # this loop sets in one scenario each "i" feature assumption
+            # k is the config input key "loads_t.p_set"
+            # v is the lower and upper bound [0.8,1.3], that was used for scenarios
+            # i, j interation number to pick values of experimental setup
+            # Example: n.loads_t.p_set = n.loads_t.p_set * scenarios[0,0]
+            exec(f"n.{k} = n.{k} * {scenarios[i,j]}")
+            logger.info(f"Scaled n.{k} by factor {scenarios[i,j]} in the {i} scenario")
+            j = j + 1
+
+    if OPTIONS.get("method") == "any_chance_store_test":
+        for k, _ in PYPSA_FEATURES.items():
+            type = k.split(".")[0]  # "stores", "generators", ...
+            feature = k.split(".")[1]  # "capital_cost", "efficiency", ...
+
+        if type == "stores":
```

Comment:
add a TODO for other cases.
If easy to include a default case may be interesting but no priority to that

Reply:
Done